jdaviz/configs/imviz/plugins/orientation/orientation.py

```diff
@@ -406,10 +427,15 @@ def _on_refdata_change(self, msg={}):
             elif not len(viewer.data()):
                 self.link_type_selected = link_type_msg_to_trait['pixels']
 
+            if msg._data.label not in self.orientation.choices:
+                return
+
+            self.orientation.selected = msg._data.label
+
             # we never want to highlight subsets of pixels within WCS-only layers,
             # so if this layer is an ImageSubsetLayerState on a WCS-only layer,
             # ensure that it is never visible:
-            for layer in viewer.state.layers:
+            for layer in self.viewer.selected_obj.state.layers:

```

Comment:
hmmm, I think this should be the same as `viewer` defined above (with or without the suggestion to use `viewer.selected_obj` there as well).